src/node.h

```diff
@@ -1548,24 +1563,10 @@ void RegisterSignalHandler(int signal,
                            bool reset_handler = false);
 #endif  // _WIN32
 
-// Configure the layout of the JavaScript object with a cppgc::GarbageCollected
-// instance so that when the JavaScript object is reachable, the garbage
-// collected instance would have its Trace() method invoked per the cppgc
-// contract. To make it work, the process must have called
-// cppgc::InitializeProcess() before, which is usually the case for addons
-// loaded by the stand-alone Node.js executable. Embedders of Node.js can use
-// either need to call it themselves or make sure that
-// ProcessInitializationFlags::kNoInitializeCppgc is *not* set for cppgc to
-// work.
-// If the CppHeap is owned by Node.js, which is usually the case for addon,
-// the object must be created with at least two internal fields available,
-// and the first two internal fields would be configured by Node.js.
-// This may be superseded by a V8 API in the future, see
-// https://bugs.chromium.org/p/v8/issues/detail?id=13960. Until then this
-// serves as a helper for Node.js isolates.
-NODE_EXTERN void SetCppgcReference(v8::Isolate* isolate,
-                                   v8::Local<v8::Object> object,
-                                   void* wrappable);
+NODE_DEPRECATED("Use v8::Object::Wrap()",
+                NODE_EXTERN void SetCppgcReference(v8::Isolate* isolate,
```

Comment:
I wonder if we can simply remove this, as the V8 update that contains v8::Object::Wrap() is bound to be semver-major anyway...

Reply:
cc @targos

Reply:
hmm, I guess an argument for keeping this would be, addons can use this API to ensure that they are able to allocate cppgc-managed objects in multiple different Node.js versions without code changes to adapt to whether wrapper-descriptor-based tracing is deprecated on a version that it is loaded (even if that involves recompilation). It doesn't cost us much to maintain this thing wrapper for a while anyway?

Reply:
The compatiblity layer is fairly simple and I don't see the harm to keep this for a while before removing it completely.

Reply:
Makes sense, I updated the PR to document what this is for and reverted the tests to keep using it.